integration/vm_factory/vm_templating_test.sh

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
+#
+# Copyright (c) 2018 HyperHQ Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# This test will perform several tests to validate kata containers with
+# vm templating.
+
+set -e -x
+
+cidir=$(dirname "$0")
+
+source "${cidir}/../../metrics/lib/common.bash"
+
+# Environment variables
+IMAGE="${IMAGE:-busybox}"
+CONTAINER_NAME=$(random_name)
+PAYLOAD_ARGS="${PAYLOAD_ARGS:-tail -f /dev/null}"
+
+# Timeout is the duration of this test (seconds)
+# We want to stress the agent for a significant
+# time (approximately running for two days)
+timeout=186400
+start_time=$(date +%s)
+end_time=$((start_time+timeout))
+
+if [ -f ${DEFAULTCONFIG} ]; then
+	runtime_config_path=${DEFAULTCONFIG}
+else
+	runtime_config_path=${PKGDEFAULTCONFIG}
+fi
+
+template_tmpfs_path="/run/vc/vm/template"
+
+enable_vm_template_config() {
+	echo "enable vm template config ${runtime_config_path}"
+	sudo sed -i -e 's/^#\(enable_template\).*=.*$/\1 = true/g' "${runtime_config_path}"
+}
+
+disable_vm_template_config() {
+	echo "disable vm template config"
+	sudo sed -i -e 's/^\(enable_template\).*=.*$/#\1 = true/g' "${runtime_config_path}"
+}
+
+init_vm_template() {
+	echo "init vm template"
+	sudo -E PATH=$PATH kata-runtime factory init
```

Comment:
centos and fedora do not find `kata-runtime` when using sudo (PATH problem it seems). We have hit this before and we have use the full path of the binary using:
```
kata_runtime_bin=$(command -v kata-runtime)
sudo -E  "$kata_runtime_bin" factory init
```